feat: improve the swap quote sort UI (#9639)

premiumjibles opened the pull request:
## Description

This implements the sorting UI adjustments in issue (#9639). Modifying the sort selector UI inside the trade settings popover. As well as introducing another sort selector inside the quote list. 

## Issue (if applicable)

This doesn't close the relevant issue as it's only a sub task
closes #9684 

<!-----------------------------------------------------------------------------
If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.
------------------------------------------------------------------------------>

## Risk

From what I can see, the components modified here are only used in one spot. Shown below 

![image](https://github.com/user-attachments/assets/1c26881e-ecf4-47e9-be24-116e51e7fca2)

So risk is only that this area will render strangely.

Some superficial adjustments were made to the WithBackButton component which is used in a number of locations. But the change was purely structural. So the risk there should be extremely low.

> High Risk PRs Require 2 approvals

<!-----------------------------------------------------------------------------
Outline the scope of your changes and the risk associated with them. You must use your discretion as an engineer to determine the potential impact of your changes.

WARNING: If your PR introduces a new on-chain transaction or modifies an existing one, please add the "High-Risk" label to this PR and ask for 2 reviewers to approve it before merging.

E.g. an upgrade to `hdwallet` or core state management would be considered higher risk, and might require a full regression test. UI or isolated view changes, or something behind a feature flag may have near zero risk. Small bug fixes might require testing isolated to the specific fix.
------------------------------------------------------------------------------>

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

## Testing

Test using both sort elements shown above to rearrange the quote list. 

Test rendering with a variety of screen sizes.

Test sort selection state is synced across both elements.

<!-----------------------------------------------------------------------------
We treat every PR to be merged with the same scrutiny as if we were merging directly to production.

Your PR will not be merged if you do not complete the sections below for our engineering and operations teams to test.
------------------------------------------------------------------------------>

### Engineering

For the quote selector to pop up. You must populate a number inside the Trade/Bridge component.

<!-----------------------------------------------------------------------------
Include sufficient information here for an engineer to test your PR. This may include how to test locally, in a built environment, changes to infrastructure etc.
------------------------------------------------------------------------------>

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

<!-----------------------------------------------------------------------------
If your changes have a user-facing impact, describe how a non-technical QA team can functionally test your changes in a preview environment.

If they are not user-facing please describe how to test for any regressions that may occur.
------------------------------------------------------------------------------>

## Screenshots (if applicable)

![image](https://github.com/user-attachments/assets/6b5e1c3f-6a09-4f51-90d1-5a1e8e1228a0)

![image](https://github.com/user-attachments/assets/9a708f50-6039-4eb0-b1d6-86dbfb9b0d72)

### Other notes
* Note that there is a separate bug here that exists in production where the trade popover settings will randomly grow and shrink in width based on changes to the quote list. 
* This is functionally ready but is still waiting on final sort copy and translations for the tooltips and info.
* Some minor deviations where made from the designs in the name of consitency. E.g. consistent info text between both sort elements and consistent ordering of sort options.

premiumjibles commented on src/components/MultiHopTrade/components/SharedTradeInput/QuoteSortSelector.tsx:
```diff
@@ -1,131 +1,51 @@
-  const translate = useTranslate()
   const currentSortOption = useAppSelector(tradeQuoteSlice.selectors.selectQuoteSortOption)
 
-  const currentSortType = useMemo(() => {
```
Attempted to simplify the below code. I couldn't see a reason for the introduction of an additional SortType enum rather than just using QuoteSortOption directly.

NeOMakinG reviewed (changes requested):
Almost perfect!

Only one blocker related to the light theme:
<img width="235" alt="image" src="https://github.com/user-attachments/assets/5198823e-f829-492b-86f8-c9989b943af8" />

NeOMakinG commented on src/components/MultiHopTrade/components/QuoteList/QuoteList.tsx:
```diff
@@ -13,14 +33,48 @@ type QuoteListProps = {
+          <Heading fontSize='md'>{translate('trade.availableSwappers')}</Heading>
+        </Flex>
+        <Flex alignItems='center' gap={2}>
+          <Text color='whiteAlpha.500' fontWeight='normal' fontSize='sm'>
```
This one also needs to use `useColorModeValue`

premiumjibles commented:
Great catch @NeOMakinG. I've fixed that up and noted for future to make sure I handle dark/light mode color switching.

NeOMakinG reviewed (approved):
Seems like it does the do!

https://jam.dev/c/1333c06e-f3f3-44ec-bc27-115e2a23f86a

Pre-stamping so you're not blocked by the product copies changes!

Well done @premiumjibles

premiumjibles commented:
@gomesalexandre I've added two commits. One to pull in your changes and one to remove tooltips. 

https://jam.dev/c/79580a42-3d59-4cf6-8391-1aa50352665e

NeOMakinG commented on src/components/MultiHopTrade/components/QuoteList/QuoteList.tsx:
```diff
@@ -13,14 +34,50 @@ type QuoteListProps = {
+    [currentSortOption, quoteSortOptions],
+  )
+
+  const sortByTextColor = useColorModeValue('blackAlpha.500', 'whiteAlpha.500')
```
❤️

premiumjibles commented:
> I seen a weird thing while clicking on options, the text is having a small bump probably because of the size of the icon, it's pretty nitpick but if you want it to be perfect, feel free to fix it at your own discretion: https://jam.dev/c/2611d82b-5e22-4f53-bfc6-0aa11f2e0e3b
> 
> it's probably just a matter of icon height!
> 
> Everything else looks good to me again!
> 
> https://jam.dev/c/15a50b72-8cf8-4f7c-bbcd-b68ba114873f

Great spot @NeOMakinG thanks for that. You're right the icon is introducing a few more pickels and the text isn't centring properly with it. I've fixed it and renamed buttonConf -> quoteSortOpt. I try to avoid variable names that are only one `s` off from another variable name to avoid accidentally using one when you meant the other.
